### PR TITLE
[Dogfooding] Enforce IDE0007 (prefer "var") as build warning in IDE projects

### PR DESCRIPTION
**Pull request by mavasani**
Only manual changes were done in https://github.com/dotnet/roslyn/commit/b884a4d63f08cc76f278fd7264a3a32e71c791a4, https://github.com/dotnet/roslyn/commit/8db8f7d9e50e9f8c1cc3195dc2deccb50d5bf6af and https://github.com/dotnet/roslyn/commit/bbb005e07d8c7486ca1e897ce383645445243017

```ini
[src/{Analyzers,CodeStyle,Features,Workspaces}/**/*.{cs,vb}]

# Prefer "var" everywhere
dotnet_diagnostic.IDE0007.severity = warning
csharp_style_var_for_built_in_types = true:warning
csharp_style_var_when_type_is_apparent = true:warning
csharp_style_var_elsewhere = true:warning
```

### CODE REVIEW

File: src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs [hunk 1]
```diff
@@ -560,7 +560,10 @@ private static bool IsParamsArrayExpanded(SemanticModel semanticModel, SyntaxNod
             }
             else
             {
+#pragma warning disable IDE0007 // Use implicit type - Using 'var' causes "error CS8506: No best type was found for the switch expression"
```
Comment (mavasani) [inline src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs:563]: @allisonchou I thought you had fixed a similar issue earlier?

Another thing to note - this IDE0007 violation only shows up in the IDE, but does not show up in the build. I am not sure about the reason for it.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs:563]: we def need to file this bug.

**Comment by mavasani** [comment]
@dotnet/roslyn-infrastructure Is `Windows_Desktop_Unit_Tests debug_64` blocked? I see it is timing out for almost all PRs as it seems to hit the 90 min limit when it is nearing completion. Does the timeout need to be increased?

File: src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs:563]: Filed https://github.com/dotnet/roslyn/issues/44507

**Pull request merged**